ext/dynamichmc.jl

```diff
@@ -0,0 +1,86 @@
+###
+### DynamicHMC backend - https://github.com/tpapp/DynamicHMC.jl
+###
+
+import DynamicHMC
+
+# Wraps DynamicHMC as an AbstractSampler
+struct DynamicNUTS{S<:DynamicHMC.NUTS} <: AbstractSampler
+    sampler::S
+end
+
+DynamicNUTS() = DynamicNUTS(DynamicHMC.NUTS())
+externalsampler(spl::DynamicHMC.NUTS) = ExternalSampler(DynamicNUTS(spl))
```

Comment:
Could we just use
```suggestion
externalsampler(spl::DynamicHMC.NUTS) = DynamicNUTS(spl)
```
?

Reply:
The problem comes from the fact that `ExternalSampler `expects an `AbstractMCMC.AbstractSampler` but DynamicHMC doesn't follow that interface. So first I have to create a wrapper for DynamicHMC.NUTS that turns it into an `AbstractMCMC.AbstractSampler`  and then apply the `ExternalSampler` wrapper to use the external sampler interface.

Ideally, we make a PR on DynamicHMC and make the samplers follow the interface but we don't own that repo.

Reply:
But do we actually want a `ExternalSampler` type here? Can't we always dispatch purely on `DynamicNUTS`? It's owned by us so there would be no type piracies involved.

Reply:
You are correct. We own `DynamicNUTS `and we could dispatch on it. 
However, I think the current approach is closer to what we would like to do if we had access to DynamicHMC and preserves the logic of the ExternalSampler.

Reply:
But as long as DynamicHMC does not implement the AbstractMCMC interface (and it's unclear if DynamicHMC ever wants to support AbstractMCMC) there's arguably no external sampler involved here. IMO also `sample(..., DynamicNUTS(), ...)` is a bit more convenient than `sample(...., externalsampler(DynamicNUTS()), ...)`.

Reply:
@yebai do you want to break the stale mate? I see the points of @devmotion, so I am happy to go down this route if we all agree.